### PR TITLE
Remove dependency on commons-io and fest-assert

### PR DESCRIPTION
**Pull request by dbmeneses**

### CODE REVIEW

File: api/src/main/java/org/sonarsource/scanner/api/internal/TempCleaning.java [hunk 1]
```diff
@@ -34,32 +35,47 @@
...
+      List<Path> files = Files.list(tempDir)
+        .filter(p -> p.getFileName().toString().startsWith("sonar-runner-batch"))
+        .filter(p -> lastModifiedTime(p) < cutoff)
+        .collect(Collectors.toList());
+
+      for (Path file : files) {
+        Utils.deleteQuietly(file);
+      }
+      logger.debug("Temp cleaning done");
+    } catch (IOException e) {
```
Comment (SonarTech) [inline api/src/main/java/org/sonarsource/scanner/api/internal/TempCleaning.java]: MAJOR Either log or rethrow this exception. [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS1166)

File: api/src/main/java/org/sonarsource/scanner/api/internal/TempCleaning.java [hunk 2]
```diff
@@ -34,32 +35,47 @@
...
+      logger.warn("Failed to clean files in " + tempDir.toString());
+    }
+  }
+
+  private long lastModifiedTime(Path file) {
```
Comment (SonarTech) [inline api/src/main/java/org/sonarsource/scanner/api/internal/TempCleaning.java]: MAJOR Make "lastModifiedTime" a "static" method. [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS2325)

File: api/src/main/java/org/sonarsource/scanner/api/internal/TempCleaning.java [hunk 3]
```diff
@@ -34,32 +35,47 @@
...
+    try {
+      return Files.getLastModifiedTime(file).toMillis();
+    } catch (IOException e) {
```
Comment (SonarTech) [inline api/src/main/java/org/sonarsource/scanner/api/internal/TempCleaning.java]: MAJOR Either log or rethrow this exception. [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS1166)

**Comment by SonarTech** [comment]
SonarQube analysis reported 3 issues
* MAJOR 3 major

Watch the comments in this conversation to review them.

File: api/src/main/java/org/sonarsource/scanner/api/internal/TempCleaning.java [hunk 4]
```diff
@@ -34,32 +33,44 @@
...
+      logger.debug("Temp cleaning done");
+    } catch (IOException e) {
+      logger.warn("Failed to clean files in " + tempDir.toString());
+    }
+  }
+
+  private static long lastModifiedTime(Path file) {
+    try {
+      return Files.getLastModifiedTime(file).toMillis();
+    } catch (IOException e) {
```
Comment (SonarTech) [inline api/src/main/java/org/sonarsource/scanner/api/internal/TempCleaning.java]: MAJOR Either log or rethrow this exception. [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS1166)

File: api/src/main/java/org/sonarsource/scanner/api/internal/TempCleaning.java [hunk 5]
```diff
@@ -34,32 +33,44 @@
...
-    for (File file : files) {
-      FileUtils.deleteQuietly(file);
+    try {
+      Files.list(tempDir)
+        .filter(p -> p.getFileName().toString().startsWith("sonar-runner-batch"))
+        .filter(p -> lastModifiedTime(p) < cutoff)
+        .forEach(Utils::deleteQuietly);
+
+      logger.debug("Temp cleaning done");
+    } catch (IOException e) {
```
Comment (SonarTech) [inline api/src/main/java/org/sonarsource/scanner/api/internal/TempCleaning.java]: MAJOR Either log or rethrow this exception. [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS1166)

**Comment by SonarTech** [comment]
SonarQube analysis reported 2 issues
* MAJOR 2 major

Watch the comments in this conversation to review them.

File: api/src/main/java/org/sonarsource/scanner/api/internal/TempCleaning.java [hunk 6]
```diff
@@ -34,32 +33,44 @@
...
+      logger.warn("Failed to clean files in " + tempDir.toString());
```
Comment (henryju) [inline api/src/main/java/org/sonarsource/scanner/api/internal/TempCleaning.java]: Is there a way to show stack trace, at least in debug mode ?

Comment (dbmeneses) [inline api/src/main/java/org/sonarsource/scanner/api/internal/TempCleaning.java]: we would need to discover logging level and also serialize the stack into a string. For simplicity I will add the exception message, which is usually self-explanatory in IOExceptions.

**Pull request merged**